src/VM/Handlers/Opcode8046Handler.cpp

```diff
@@ -35,9 +37,21 @@ Opcode8046Handler::Opcode8046Handler(VM* vm) : OpcodeHandler(vm)
 
 void Opcode8046Handler::_run()
 {
-    Logger::debug("SCRIPT") << "[8046] [*] - value (change sign)" << std::endl;
-    auto value = _vm->popDataInteger();
-    _vm-> pushDataInteger(-value);
+    Logger::debug("SCRIPT") << "[8046] [*] op_negate" << std::endl;
+    auto value = _vm->dataStack()->pop();
+    if (value->type() == VMStackValue::TYPE_INTEGER) 
+    {
+        _vm->pushDataInteger(- (dynamic_cast<VMStackIntValue*>(value))->value());
+    }
+    else
+    if (value->type() == VMStackValue::TYPE_FLOAT) 
+    {
+        _vm->pushDataFloat(- (dynamic_cast<VMStackFloatValue*>(value))->value());
+    }
+    else
+    {
```

Comment:
No, that's not a good idea. Currently we have *levels* and *subsystems* for log events. Different static methods in Logger indicate different event levels. All script warnings are *info* at best, but may use "scripts" subsystem. See this: https://github.com/falltergeist/falltergeist/issues/288.